### ISSUE TITLE
gitkit.Request.RepoName isn't populated correctly for subpath repos

### ISSUE THREAD

**Issue by cweagans** [sosedoff/gitkit#11]
if I push to someorg/somerepo, RepoName is set to "someorg" instead of "somerepo". RepoName should probably be set to the last segment in the path.

RepoPath is set properly in this scenario.

**Comment by sosedoff** [comment]
Yeah, seems like the request should include `Namespace` and `RepoName`. 
So it would work like this:

- `http://gitkit/repo` -> namespace: "", repo: "repo"
- `http://gitkit/org/repo` -> namespace "org", repo: "repo"
- `http://gitkit/org/foo/bar/repo` -> namespace: "org/foo/bar", repo: "repo"

Would that work for you?

**Comment by sosedoff** [comment]
What's your repos dir structure look like? Something like `/path/namespace/repo` ?

**Comment by cweagans** [comment]
:party:
